src/driver-sinowealth.c

```diff
@@ -404,6 +410,30 @@ sinowealth_commit(struct ratbag_device *device)
 	int rc;
 	uint8_t dpi_enabled = 0;
 
+	char reported_rate;
+	switch (profile->hz) {
+		case 125:
+			reported_rate = 0x01;
+			break;
+		case 250:
+			reported_rate = 0x02;
+			break;
+		case 500:
+			reported_rate = 0x03;
+			break;
+		case 1000:
+			reported_rate = 0x04;
+			break;
+		default:
+			log_info(device->ratbag, "invalid report rate (%d); falling back to 1000 hz.\n", profile->hz);
+			reported_rate = 0x04;
+			break;
+	}
+
+	/* Update report rate */
+	config->config &= ~0x0f;
+	config->config |= reported_rate;
```

Comment:
Can this be made in a more optimal way?

Reply:
I checked it it assembly, and it's better when it's done separately on two lines. Still, 0x0f mask should be replaced with 0b111 (0x7).